python/util.py

```diff
@@ -0,0 +1,257 @@
+from functools import wraps
+from collections import namedtuple
+from itertools import starmap
+from operator import add, sub, eq
+
+
+def cache(method):
+    attr_name = '_' + method.__name__
```

Comment:
I think it's ok - you can set and get  `__name` attributes just fine that don't get name munged. The only constraint then is that you can't use them with slots, but all the other alternatives I tried seemed more messy. My next favourite was `___`, but that felt like it was getting mental. Open to suggestions!